<?xml version="1.0"?>
<launch>  
  <arg name="joy_config" default="my_xbox" />
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="config_filepath" default="$(find twist_vehicle_cmd)/config/$(arg joy_config).config.yaml" />
  
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="20" />
  </node>

  <node pkg="teleop_twist_joy" name="teleop_twist_joy" type="teleop_node">
    <rosparam command="load" file="$(arg config_filepath)" />
  </node>
  

  <node pkg="twist_vehicle_cmd" type="twist_vehicle_mapper" name="twist_vehicle_mapper_node" output="screen">
	<param name="throttle_max" value="50.0" type="double"/>
    <param name="steering_max" value="40.0" type="double"/>
	<param name="twist_v_max" value="2.0" type="double"/>
	<param name="twist_w_max" value="2.0" type="double"/>
  </node> 
	
</launch>
